### PR TITLE
Accept filters in list containers

### PR DESCRIPTION
**Pull request by carlossg**
PR on top of #260, to test with container labels

### CODE REVIEW

**Comment by carlossg** [comment]
The filters query param in Docker API is a pita, it is a json representation of the filters (`Map<String, String[]>`) but needs to be URL encoded for jersey to work, or it barfs on the curly braces `{` because it thinks it is a jersey template.

File: src/test/java/com/github/dockerjava/core/command/ListContainersCmdImplTest.java [hunk 1]
```diff
@@ -96,7 +99,22 @@ public void testListContainers() throws DockerException {
 
         Container container2 = filteredContainers.get(0);
         assertThat(container2.getCommand(), not(isEmptyString()));
-        assertThat(container2.getImage(), startsWith(testImage + ":"));
+        assertThat(container2.getImage(), startsWith(testImage));
+
+        // list with filter by label
+        filteredContainers = dockerClient.listContainersCmd().withShowAll(true)
```
Comment (marcuslinke) [inline src/test/java/com/github/dockerjava/core/command/ListContainersCmdImplTest.java]: This specific test doesn't work in my local test env (docker api 1.18).

**Comment by marcuslinke** [comment]
Seems the label filtering is somehow broken with docker 1.7.0/1.19 also. It finds containers that haven't a label at all. An appropriate log example:

```
23:08:38.856 DEBUG c.g.d.jaxrs.DockerCmdExecFactoryImpl - 29 * Sending client request on thread main
29 > GET https://192.168.59.103:2376/containers/json?filters=%7B%22listcontainertest%22:%5B%22docker-java%22%5D%7D&all=1&size=0
29 > Accept: application/json

23:08:39.121 DEBUG c.g.d.jaxrs.DockerCmdExecFactoryImpl - 30 * Client response received on thread main
30 < 200
30 < Content-Length: 538
30 < Content-Type: application/json
30 < Date: Thu, 09 Jul 2015 21:08:37 GMT
[{"Id":"c10fa26069f9421daec0ad15c5e02626e781a7ce48f6293e90b6b033b779ec75","Names":["/gloomy_almeida"],"Image":"busybox","Command":"echo","Created":1436476116,"Ports":[],"Labels":{"listcontainertest":"docker-java"},"Status":"Exited (0) Less than a second ago"},{"Id":"f993b8837846a66fcb590c40bc0b511efaeff036870a12ad13c0f6d8828dce09","Names":["/prickly_payne"],"Image":"ee151a08220c9dee6f21dc1db29e53a09d11fa5dac21f8ec0fb46fea8c1f9bd8","Command":"./cmd.sh","Created":1436475851,"Ports":[],"Labels":{},"Status":"Exited (0) 4 minutes ago"}]
```

**Comment by carlossg** [comment]
Yes, seems I misunderstood the docs. I have debugged the daemon logs and updated the implementation to work with `withLabel("test")` or `withLabel("test=docker-java")`
Also improved the test

**Comment by marcuslinke** [comment]
@carlossg Thanks for your contribution!

**Pull request merged**